modules/juce_gui_extra/native/juce_win32_WebBrowserComponent.cpp

```diff
@@ -74,13 +76,68 @@ class WebBrowserComponent::Pimpl   : public ActiveXControlComponent
                 auto* owner = dynamic_cast<WebBrowserComponent*> (getParentComponent());
                 jassert (owner != nullptr);
 
-                auto handler = new EventHandler (*owner);
+                auto handler = new EventHandler (*owner, *this);
                 connectionPoint->Advise (handler, &adviseCookie);
                 handler->Release();
             }
         }
     }
 
+    void loadHTMLString(const String& htmlString, const String& /*baseURLString*/)
+    {
+        if (htmlSource)
+        {
+            delete[] htmlSource;
+        }
+        htmlSource = new wchar_t[htmlString.length() + 1];

```

Comment:
When `WebBrowserComponent::Pimpl` is destroyed, it seems like `htmlSource` will be leaked. Would it be possible to use `std::unique_ptr<wchar_t[]>` or `std::vector<wchar_t>` instead? This would also avoid double-delete if `loadHTMLSource` is called after `loadHTMLString`.

Reply:
Okay, I made htmlSource a juce::String and made the lifetime of `new wchar_t[htmlSource.length() + 1];` contained within `loadHTMLSource()`

Reply:
The lifetime is still spread across two functions (loadHTMLString and loadHTMLSource). Why not just use std::vector<wchar_t>?

Reply:
Lifetime of what? With my latest commit, htmlSource is now a class member variable juce::String. The new "context" allocation/deallocation is happening all within the scope of loadHTMLSource()